### ISSUE TITLE
Needed a list of mismatch rows which is common in 2 DataFrames and having only the columns which have mismatch values.

### ISSUE THREAD

**Issue by sujitsatapathy84** [capitalone/datacompy#135]
Hi,
I Needed a list of mismatch rows which is common in 2 DataFrames and having only the columns which have mismatch values.
For ex, I have df1 and df2 having 70 rows in common but there is 16 rows with mismatches for some columns. So I wanted those 16 rows with only the columns having the mismatches and do not need the columns which are matching.

I tried using compare.all_mismatch(), it gave me the 16 rows which are having mismatches but it had all the columns both matched and unmatched. Since there are many columns, it becomes difficult to find the unmatched column, in this way. So needed a solution.

Thanks in Advance.
Regards
Sujit

**Comment by fdosani** [comment]
You can do something like this

setup some example data

```python
from io import StringIO
import pandas as pd
import datacompy

data1 = """acct_id,dollar_amt,name,float_fld,date_fld
10000001234,123.45,George Maharis,14530.1555,2017-01-01
10000001235,0.45,Michael Bluth,1,2017-01-01
10000001236,1345,George Bluth,,2017-01-01
10000001237,123456,Bob Loblaw,345.12,2017-01-01
10000001238,1.05,Lucille Bluth,,2017-01-01
10000001238,1.05,Loose Seal Bluth,,2017-01-01
"""

data2 = """acct_id,dollar_amt,name,float_fld
10000001234,123.4,George Michael Bluth,14530.155
10000001235,0.45,Michael Bluth,
10000001236,1345,George Bluth,1
10000001237,123456,Robert Loblaw,345.12
10000001238,1.05,Loose Seal Bluth,111
"""

df1 = pd.read_csv(StringIO(data1))
df2 = pd.read_csv(StringIO(data2))


compare = datacompy.Compare(
    df1,
    df2,
    join_columns='acct_id',  #You can also specify a list of columns
    abs_tol=0.0001,
    rel_tol=0,
    df1_name='original',
    df2_name='new')
```

Then find all the mismatch columns and save them in a list:

```python
column_names_mismatch = sum([[x["column"]+"_df1",x["column"]+"_df2"] for x in compare.column_stats if x["unequal_cnt"] > 0], [])
column_names_mismatch

>  ['name_df1', 'name_df2', 'float_fld_df1', 'float_fld_df2']
```

Finally you can subset `all_mismatch`

```python
compare.all_mismatch()[column_names_mismatch]
>
         name_df1              name_df2  float_fld_df1  float_fld_df2
0  George Maharis  George Michael Bluth     14530.1555      14530.155
1   Michael Bluth         Michael Bluth         1.0000            NaN
2    George Bluth          George Bluth            NaN          1.000
3      Bob Loblaw         Robert Loblaw       345.1200        345.120
4   Lucille Bluth      Loose Seal Bluth            NaN        111.000
```

Hopefully this helps. 
TLDR: you can do this with some work but isn't support with a single function or feature.

**Comment by sujitsatapathy84** [comment]
Thanks a lot this works but we would also need the join column in the final output, so as to identify the exact row based on the join column having mismatch. The column_stats has join column with unequal_cnt=0, how can we add the join column in the final output?

Thanks 
Sujit

**Comment by fdosani** [comment]
Just added it to your list of columns `column_names_mismatch`. If you know the name?

Please don’t take this the wrong way but it might be helpful to read the docs or take a pandas or python tutorial. 

Some of the questions here are more just implementation details vs issues or package problems.
